src/m_misc.c

```diff
@@ -540,6 +540,11 @@ void NormalizeSlashes(char *str)
             {
                 p++;
             }
+            // Remove trailing slash before PATH_SEPARATOR
+            if (*p == PATH_SEPARATOR)
+            {
+                str--;
```

Comment:
Hmm, why do we need this?

Reply:
Also we can use `M_DirName` function.

Reply:
> Hmm, why do we need this?

Because on my system
```
$ echo $XDG_DATA_DIRS
/usr/share/gnome:/usr/local/share/:/usr/share/
```
and thus by splitting at `':'` (i.e. `PATH_SEPARATOR`) we get a trailing slash from the `/usr/local/share/:` part.

Reply:
Ok, got it.

Reply:
> Also we can use `M_DirName` function.

Nope, this doesn't work. It turns `/usr/share/games/doom` into `/usr/share/games`.